criu/seize.c

```diff
@@ -967,6 +968,54 @@ static int cgroup_version(void)
 	return -1;
 }
 
+static int pause_devices_in_cgroup(void)
+{
+	char buffer[4096];
+	char procs_path[PATH_MAX];
+	ssize_t bytes_read;
+	pid_t pid;
+	int procs_fd;
+
+	/* Open the cgroup.procs file */
+	snprintf(procs_path, sizeof(procs_path), "%s/%s", opts.freeze_cgroup, "cgroup.procs");
+	procs_fd = open(procs_path, O_RDONLY);
+	if (procs_fd == -1) {
+		pr_perror("Failed to open cgroup.procs file");
+		return -1;
+	}
+
+	/* Read cgroup.procs into a buffer */
+	while ((bytes_read = read(procs_fd, buffer, sizeof(buffer) - 1)) > 0) {
```

Comment:
you must understand the last line in the buffer can be incomplete and you need to handle it properly